modules/transactions/src/main/java/org/apache/ignite/internal/tx/LockException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.tx;
+
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+
+/**
+ * This exception is thrown when lock cannot be acquired due to conflict.
+ */
+public class LockException extends IgniteInternalCheckedException {
+    /**
+     * @param msg The message.
+     */
+    public LockException(String msg) {
+        super(msg);
+    }
+
+    /**
+     * @param waiter Conflicting waiter.
+     */
+    public LockException(Waiter waiter) {
+        super("Lock can't be taken because of the conflict with " + waiter);
```

Comment:
Should we use wording consistent with Ignite 2.x? `"Failed to acquire lock due to a conflict with: " + waiter` ?